Tools/src/MuonPogTree.h

```diff
@@ -19,22 +22,35 @@ namespace muon_pog {
     
     ClassDef(GenInfo,1)
   };
-
+  
   class GenParticle {
   public:
-    Int_t pdgId; // PDG identifier
+    
+    GenParticle(){};
+    virtual ~GenParticle(){};
+    
+    Int_t pdgId;  // PDG identifier
     Int_t status; // MC status
+    bool IsPrompt; // GenStatusFlag: Lepton is NOT coming from hadron or tau decay 
     Float_t energy; // energy [GeV]
     Float_t pt; // pt [GeV]
     Float_t eta; // eta
     Float_t phi; // phi
     Float_t vx; // x coordinate of production vertex [cm]
-    Float_t vy;// y coordinate of production vertex [cm]
-    Float_t vz;// z coordinate of production vertex [cm]
+    Float_t vy; // y coordinate of production vertex [cm]
+    Float_t vz; // z coordinate of production vertex [cm]
     std::vector<Int_t> mothers; // vector of indices of mothers
 
-    GenParticle(){};
-    virtual ~GenParticle(){};
+    void SetFlags(reco::GenStatusFlags statusflags){
+      _flags.clear();
+      for (unsigned int fl = 0; fl < 15; fl++) 
+	_flags.push_back(statusflags.flags_[fl]);      
+    }
+    
+    bool flags( int flag ) { return _flags[flag]; }
+    
+  private:
+    std::vector<bool> _flags;    
```

Comment:
Why do we need a private _flags + its accessor?
The rest of the class has public variables (like a struct), could you simply move to a public : std::vector<bool> flags; ?